### PR TITLE
Fix for SPI and CRC reg changes in stm32-data

### PR DESCRIPTION
**Pull request by taunusflieger**
Fix for issue https://github.com/embassy-rs/stm32-data/issues/459

### CODE REVIEW

File: embassy-stm32/src/crc/v1.rs [hunk 1]
```diff
@@ -32,20 +32,32 @@ impl<'d> Crc<'d> {
     /// Feeds a word to the peripheral and returns the current CRC value
     pub fn feed_word(&mut self, word: u32) -> u32 {
         // write a single byte to the device, and return the result
+        #[cfg(not(crc_v1))]
+        PAC_CRC.dr32().write_value(word);
```
Comment (dlaw) [inline embassy-stm32/src/crc/v1.rs:36]: At least on STM32G4, any write to dr32 writes _four_ bytes into the CRC (hence the suggestive name). (Don't ask how long it took for me to figure this out... the reference manual hardly explains it at all :tired_face:)

If only a single byte write is intended, I suggest to write `PAC_CRC.dr8()` instead. If a four-byte write is intended, then the comment 1 line above should be updated.

Comment (taunusflieger) [reply embassy-stm32/src/crc/v1.rs:36]: Good catch will rework - unfortunately there are other breaking changes (inc. tests) which need to be fixed all together to get a successful CI run - I'm not sure if I find the time to address them.

Comment (taunusflieger) [reply embassy-stm32/src/crc/v1.rs:36]: To me it looks more like a question if that comment is correct - the function is called `feed_word()` why should it write a byte? It takes a `u32` - a word. Shouldn't we have another function to write a byte?

Comment (dlaw) [reply embassy-stm32/src/crc/v1.rs:36]: Indeed, the comment is not consistent with the name of the function. Looking at crc/v2v3.rs, there are separate functions for `feed_word()` and `feed_byte()`.

File: embassy-stm32/src/adc/g4.rs [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+            reg.set_deeppwd(false);
+            reg.set_advregen(true);
+        });
+
+        blocking_delay_us(10);
+    }
+
+    fn configure_differential_inputs(&mut self) {
+        T::regs().difsel().modify(|w| {
+            for n in 0..20 {
```
Comment (idaniel86) [inline embassy-stm32/src/adc/g4.rs:173]: Hi, there is a bug here, there are max 18 channels, not 20.

It results in panic:
0.000056 ERROR panicked at /Users/xxx/.cargo/git/checkouts/stm32-data-generated-cb34dad5f3150296/687d64d/stm32-metapac/src/chips/stm32g491re/../../peripherals/adc_g4.rs:259:78:
assertion failed: n < 18usize

Comment (Dirbaio) [reply embassy-stm32/src/adc/g4.rs:173]: can you send a pull request? :pray:

**Pull request merged**